package.json

```diff
@@ -62,9 +62,11 @@
     "lint:tests": "tslint -p tsconfig.json -t stylish -c src/tests/tslint.json -e src/tests/references.ts \"src/tests/**/*.ts\"",
     "prettier": "prettier --write \"**/*.{js,ts,tsx,json,md}\"",
     "start": "npm-run-all clean copy:json watch:src",
-    "test:mocha": "mocha \"dist/src/tests/**/*.js\" --timeout 5000",
+    "test:mocha": "mocha \"dist/src/tests/**/*.js\" --timeout 5000 --reporter mocha-junit-reporter",
     "test:rules": "tslint -r dist/src --test \"tests/**\"",
-    "test": "npm-run-all clean copy:json tsc:src test:* tslint:check lint:* validate:* copy:package generate:* copy:meta copy:config-json",
+    "test": "npm-run-all clean copy:json tsc:src test:* tslint:check lint:* validate:* copy:package copy:meta generate:*",
+    "test-ci": "npm-run-all test test-junit",
```

Comment:
So in order to run on Azure Pipelines, there needs to be a _second_ run of the tests, with a different reporter? That seems off. Can we just go by the exit code of the regular `test:mocha` and `test:rules` tasks? `test:rules` won't play nicely with JUnit reporting, as that calls to the built-in TSLint test runner instead of a standard framework like Mocha. 

Suggestion: how about removing any changes to `package.json` and `package-lock.json`, and going just off the exit code of `npm run test`?

Reply:
So it looks like I was only running npm test and not npm test-ci so it wasn't running the junit tests and reporting. I should be able to just change the yml file with the script and remove the changes to package.json and package-lock.json